data/core/start.lua

```diff
@@ -19,3 +19,18 @@ package.path = DATADIR .. '/?.lua;' .. package.path
 package.path = DATADIR .. '/?/init.lua;' .. package.path
 package.path = USERDIR .. '/?.lua;' .. package.path
 package.path = USERDIR .. '/?/init.lua;' .. package.path
+
+local dynamic_suffix = MACOS and 'lib' or (WINDOWS and 'dll' or 'so')
+package.cpath = DATADIR .. '/?.' .. dynamic_suffix .. ';' .. package.cpath
+package.cpath = USERDIR .. '/?.' .. dynamic_suffix .. ';' .. package.cpath
+package.searchers[3] = function(modname)
+  local s,e = 0
+  repeat
```

Comment:
Could `package.searchpath` be used here instead?

Reply:
It absolutely could be. Good call; I'll refactor.